smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/AcceptorGenerator.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.kotlin.codegen.rendering.waiters
+
+import software.amazon.smithy.codegen.core.CodegenException
+import software.amazon.smithy.jmespath.JmespathExpression
+import software.amazon.smithy.kotlin.codegen.core.KotlinWriter
+import software.amazon.smithy.kotlin.codegen.core.RuntimeTypes
+import software.amazon.smithy.kotlin.codegen.core.withBlock
+import software.amazon.smithy.kotlin.codegen.utils.dq
+import software.amazon.smithy.model.shapes.ShapeId
+import software.amazon.smithy.waiters.*
+
+/**
+ * Renders an individual acceptor for a waiter.
+ */
+private fun KotlinWriter.renderAcceptor(wi: WaiterInfo, acceptor: Acceptor) {
+    addImport(RuntimeTypes.Core.Retries.RetryDirective)
+
+    val directive = when (acceptor.state!!) {
+        AcceptorState.SUCCESS -> "TerminateAndSucceed"
+        AcceptorState.FAILURE -> "TerminateAndFail"
+        AcceptorState.RETRY -> {
+            addImport(RuntimeTypes.Core.Retries.RetryErrorType)
+            "RetryError(RetryErrorType.ServerSide)"
+        }
+    }
+
+    when (val matcher = acceptor.matcher) {
+        is Matcher.SuccessMember -> {
+            addImport(RuntimeTypes.Core.Retries.Impl.Waiters.SuccessAcceptor)
+            write("SuccessAcceptor(RetryDirective.#L, #L),", directive, matcher.value)
+        }
+
+        is Matcher.ErrorTypeMember -> renderErrorAcceptor(wi, directive, matcher)
+        is Matcher.InputOutputMember -> renderPathAcceptor(wi, directive, true, matcher.value)
+        is Matcher.OutputMember -> renderPathAcceptor(wi, directive, false, matcher.value)
+        else -> throw CodegenException("""Unknown matcher type "${matcher::class}"""")
+    }
+}
+
+/**
+ * Render the top-level list of acceptors for a waiter.
+ */
+fun KotlinWriter.renderAcceptorList(wi: WaiterInfo) {
```

Comment:
It can't be private since it's accessed by code in **WaiterGenerator.kt**. (Originally, everything was in one file except the visitor and it was too huge so I tried to split it up.) I can make it `internal` but I don't think that alleviates the intellisense concern.

Reply:
This isn't a big deal overall, just came to mind while reading through it.

Reply:
I'll mark this as `internal` as a slight clarification of intent.